torchdata/dataloader2/communication/iter.py

```diff
@@ -264,6 +277,8 @@ def __iter__(self):
                         raise communication.iter.InvalidStateResetRequired
                     if isinstance(response, communication.messages.TerminateResponse):
                         raise communication.iter.TerminateRequired
+                    if isinstance(response, communication.messages.WorkerExceptionResponse):
+                        raise communication.iter.WorkerException(f"Exception from worker {idx}: {response.exception}")
```

Comment:
The argument of `WorkerException` should be an exception rather than string. The easiest way to chain multiple exceptions should be:
```py
raise communication.iter.WorkerException(idx) from response.exception
```
In `WorkerException`, you can do:
```py
class WorkerException:
    def __init__(self, worker_id):
        super().__init__(f"Got exception from worker {idx}")
```